src/main/java/frc/robot/commands/CrocodileDefaultCommand.java

```diff
@@ -3,33 +3,49 @@
 import java.util.function.BooleanSupplier;
 import java.util.function.DoubleSupplier;
 
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.wpilibj.XboxController;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.subsystems.CrocodileSubsystem;
 
 public class CrocodileDefaultCommand extends CommandBase {
     private final CrocodileSubsystem subsystem;
-    private final DoubleSupplier motorSpeed;
-    private final BooleanSupplier slowMovement;
+    private final DoubleSupplier trigger;
+    private final XboxController driverController;
+    private boolean beamBreakTracker = false;
+    private final Debouncer debouncer;
 
-    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier motorSpeed,
-            BooleanSupplier slowMovement) {
+    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier trigger) {
         this.subsystem = subsystem;
-        this.motorSpeed = motorSpeed;
-        this.slowMovement = slowMovement;
+        this.trigger = trigger;
+        //TODO: get driver controller port
+        //controller to use for rumble
+        driverController = new XboxController(0);
+        debouncer = new Debouncer(0.5);
         addRequirements(subsystem);
     }
 
     // Called every time the scheduler runs while the command is scheduled.
     @Override
     public void execute() {
-        if (slowMovement.getAsBoolean()) {
-            subsystem.setMotorSpeed(.4);
-        } 
-        else if (Math.abs(motorSpeed.getAsDouble()) > 0.05){
-            subsystem.setMotorSpeed(motorSpeed.getAsDouble());
+        long startTime = 0;
+        subsystem.setIntakeSpeed(trigger.getAsDouble());
+        //sets rumble if beam break is broken for 0.5 seconds and is not already rumbling
+        if(debouncer.calculate(subsystem.getBeamBreak()) && !beamBreakTracker){
+            //get current time and set rumble
+            startTime = System.currentTimeMillis();
+            driverController.setRumble(XboxController.RumbleType.kLeftRumble, 0.5);
+            driverController.setRumble(XboxController.RumbleType.kRightRumble, 0.5);
+            beamBreakTracker = true;
         }
-        else {
-            subsystem.setMotorSpeed(0);
+        if(System.currentTimeMillis() - startTime > 500){
+            //stop rumble after 500ms
+            driverController.setRumble(XboxController.RumbleType.kLeftRumble, 0);
+            driverController.setRumble(XboxController.RumbleType.kRightRumble, 0);
+        }
+        if(debouncer.calculate(!subsystem.getBeamBreak())){
```

Comment:
Why is this one negated, and the one above it on line 34 is not? Which one is correct?

Reply:
I'm not exactly sure how this debouncer works, basically, i want this if statement to execute if the beam break doesn't detect anything for 500 milliseconds

Reply:
I think your debouncer logic can be much simpler.

It's set with `kBoth`, so it will take 500ms of true inputs to actually turn true, and 500ms of false inputs to actually go back to false.

So when it turns true, start the rumble. When it turns false, stop the rumble. So that's a single if/else statement. Don't try to calc it twice with opposite values, as it is written now, since that will make it do nothing.

So I think you want something like this:

```
if (debouncer.calculate(subsystem.getBeamBreak()) {
  giveHapticFeedback.accept(0.5);
  timer.start();       
} else {
  giveHapticFeedback.accept(0);
}
```

Of course, you can still turn the rumble off with your timer, too.